src/components/PopupWithForm.js

```diff
@@ -2,19 +2,19 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
   #handleFormSubmit;
   #modalForm;
-
+  #modalSaveButton;
   #modalInputs;
   constructor(popupSelector, handleFormSubmit) {
     super(popupSelector);
 
     this.#handleFormSubmit = handleFormSubmit;
     this.#modalForm = this._modal.querySelector(".modal__form");
+    this.#modalSaveButton = this.#modalForm.querySelector(
+      ".modal__button-save"
+    );
     this.#modalInputs = this.#modalForm.querySelectorAll(".modal__form-input");
   }
-  close() {
-    super.close();
-    this.#modalForm.reset();
-  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`